Topics/Development_Process/MongoDB_in_Docker.md

```diff
@@ -0,0 +1,110 @@
+        - This corresponds to which folder in the container you want to be externally mounted
+    - The Host path can then be any folder you like. This is where the data will live in your local machine. 
+        - If you want to use this data in an another MongoDB container, you can simply remount the host path to the same container path.
+    - The environment variables for MongoDB are generally the root user and password.
+        - MONGO_INITDB_ROOT_USERNAME, MONGO_INITDB_ROOT_PASSWORD are the variables
+        - You can set the value of these to be whatever you want
+
+In a final summary, what we have done so far is setup a MongoDB server in a docker container. Following the steps in the NoSQL page about MongoDB, using MongoDB compass, you can access your database through `'localhost:27017'` and putting username and password that you have set using the environment variables.
+
+In one command line, you can use:
+
+```console
+docker run --name containername -p 27017:27017 -e MONGO_INITDB_ROOT_USERNAME=user -e MONGO_INITDB_ROOT_PASSWORD=pass mongodb/mongodb-community-server:latest
+```
+
+From what we discussed above, you can infer that -p is publish port, -e is an environment variable, and the final argument is the image. MongoDB recommends specifying the version instead of latest to ensure the version used is the same as expected. 
+
+Congratulations! You have created your first MongoDB container!
+
+### A quick summary about ports: 
+
+The publish port argument generally looks like: `XXXX:YYYY`. 
+
+The `XXXX` portion is the port that is used/exposed by the localhost. i.e. if the database was exposed on port `12345`, to access the database you would use 'mongodb://localhost:`12345`/'
+
+The `YYYY` portion is the port that is used/exposed within the docker container. This port cannot be accessed from localhost, unless published and mapped to an `XXXX` port. 
+
+Sometimes the publish port argument also has `/tcp or /http`, these signify the protocol used by that port. 
+
+There are reserved ports that cannot be used as well as unofficial ports that most people agreed to be reserved. For example, most commonly `8000 or 8080` is used for `http` servers. There is a lot of inormation that can be said about this, so for the sake of this tutorial it is common to use the same port exposed in the container for database images. For the complete list of reserved ports see: [IANA](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml).
+
+The Docker documentation goes into more detail about ports and the publish ports argument. 
+
+## Setup Alongside Application
+
+If you want to use MongoDB with another image, continuing on the example presented on the Docker page: 
+
+```yml
+version: "3.9"
+
+services:
+  app:
+    image: YOUR_IMAGE_NAME_HERE:latest
+    restart: always
+    ports:
+      - 80:80  # Expose any ports you need
+    depends_on:
+      - mongodb
+    environment: 

```

Comment:
I think this environment section should be tabbed left too

Reply:
I think the original tabbing is correct, the environment should be parented by 'app' and not services.

Reply:
Ah I see, I thought it was similar to line 94's environment that's why, my bad.